From 02f0d8de677d58bcac2fe1d2ed010389a644dc2b Mon Sep 17 00:00:00 2001
From: Emmanuel Berthier <emmanuel.berthier@intel.com>
Date: Fri, 8 Apr 2016 14:58:29 +0200
Subject: [PATCH 2/5] iio: magn: ak09911: Out of bounds access in
 regcache_flat_init

it was observed that the max_register for the regmap was not being
initialized, this could cause memory corruption issues.
initializing the value of this variable

Change-Id: I2f7ebcb17c3785f12a2e34c59f870195f732cbd0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-28402
Signed-off-by: Emmanuel Berthier <emmanuel.berthier@intel.com>
Reviewed-on: https://android.intel.com:443/492456
---
 drivers/iio/magnetometer/ak09911.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/iio/magnetometer/ak09911.c b/drivers/iio/magnetometer/ak09911.c
index f054303..84e4916 100644
--- a/drivers/iio/magnetometer/ak09911.c
+++ b/drivers/iio/magnetometer/ak09911.c
@@ -70,6 +70,8 @@
 #define AK09911_REG_CNTL2		0x31
 #define AK09911_REG_CNTL3		0x32
 
+#define AK09911_MAX_REGS		0x63
+
 #define RAW_TO_GAUSS(asa)	((((asa) + 128) * 6000) / 256)
 
 #define AK09911_CNTL2_CONTINUOUS_1_BIT	BIT(1)
@@ -490,6 +492,7 @@ static const struct regmap_config ak09911_regmap_config = {
 
 	.reg_defaults  = ak09911_reg_defaults,
 	.num_reg_defaults = ARRAY_SIZE(ak09911_reg_defaults),
+	.max_register = AK09911_MAX_REGS,
 };
 static irqreturn_t ak09911_trigger_handler(int irq, void *p)
 {
-- 
1.9.1

